src/hepattn/utils/sorting.py

```diff
@@ -0,0 +1,127 @@
+import torch
+from torch import Tensor, nn
+
+
+class Sorter:
+    def __init__(
+        self,
+        input_sort_field: str | None = None,
+        raw_variables: list[str] | None = None,
+        input_nets: nn.ModuleList | None = None,
+    ) -> None:
+        self.input_sort_field = input_sort_field
+        self.raw_variables = raw_variables or []
+        self.input_nets = input_nets or nn.ModuleList()
+
+    def sort_inputs(self, x: dict[str, Tensor]) -> dict[str, Tensor]:
+        """Sort inputs before passing to encoder for better window attention performance.
+
+        Parameters
+        ----------
+        x : dict[str, Tensor]
+            Dictionary containing embeddings and other data to be sorted.
+
+        Returns:
+        -------
+        dict[str, Tensor]
+            Sort indices for key and query dimensions.
+        """
+        self.sort_indices = {}
+
+        # Get key_embed shape for reference in sorting
+        self.num_hits = self._get_num_hits(x)
+        assert self.num_hits > 0, "key embed not found in x"
```

Comment:
I would remove the `_get_num_hits()`, it's only used once. We can just do

```
self.num_hits = x["key_embed"].shape[-2]
```

Which will nautrally raise an `KeyError` if the expected inputs are not present